lib/plausible_web/controllers/api/external_controller.ex

```diff
@@ -450,43 +450,56 @@ defmodule PlausibleWeb.Api.ExternalController do
 
   @decorate trace("ingest.geolocation")
   defp visitor_location_details(conn) do
-    result =
-      PlausibleWeb.RemoteIp.get(conn)
-      |> Geolix.lookup()
+    ip = PlausibleWeb.RemoteIp.get(conn)
 
-    country_code =
-      get_in(result, [:geolocation, :country, :iso_code])
-      |> ignore_unknown_country
+    case :locus.lookup(:geolocation, ip) do
+      {:ok, result} -> get_location_details(result)
+      _ -> get_location_details(%{})
+    end
+  end
 
-    city_geoname_id = get_in(result, [:geolocation, :city, :geoname_id])
+  defp get_location_details(geo_data) do
+    %{
+      country_code: get_country_code(geo_data),
+      subdivision1_code: get_subdivision_code(geo_data, 0),
+      subdivision2_code: get_subdivision_code(geo_data, 1),
+      city_geoname_id: get_city_geoname_id(geo_data)
+    }
+  end
 
-    subdivision1_code =
-      case result do
-        %{geolocation: %{subdivisions: [%{iso_code: iso_code} | _rest]}} ->
-          country_code <> "-" <> iso_code
+  defp get_country_code(%{}), do: ""
 
-        _ ->
-          ""
-      end
+  defp get_country_code(geo_data) do
+    geo_data
+    |> Map.get("country", %{})
+    |> Map.get("iso_code")
+    |> ignore_unknown_country()
+  end
 
-    subdivision2_code =
-      case result do
-        %{geolocation: %{subdivisions: [_first, %{iso_code: iso_code} | _rest]}} ->
-          country_code <> "-" <> iso_code
+  defp get_city_geoname_id(%{}), do: ""
 
-        _ ->
-          ""
-      end
+  defp get_city_geoname_id(geo_data) do
+    city =
+      geo_data
+      |> Map.get("city", %{})
+      |> Map.get("geoname_id", "")
 
-    %{
-      country_code: country_code,
-      subdivision1_code: subdivision1_code,
-      subdivision2_code: subdivision2_code,
-      city_geoname_id: Map.get(@city_overrides, city_geoname_id, city_geoname_id)
-    }
+    Map.get(@city_overrides, city, city)
```

Comment:
Checked `@city_overrides` in the main plausible fork to see if there were some changes since we last checked out from it; here's the only part that differs: 

```
    # Innenstadt I -> Frankfurt am Main
    6_946_225 => 2_925_533,
    # Innenstadt II -> Frankfurt am Main
    6_946_226 => 2_925_533,
    # Innenstadt III -> Frankfurt am Main
    6_946_227 => 2_925_533,
```

Reply:
Up to you if this is worth updating.

Reply:
Probably not super important to have these updated right now, I think.